src/test/java/jenkins/metrics/impl/JenkinsMetricProviderImplTest.java

```diff
@@ -136,6 +143,34 @@ public void given__a_job__when__built__then__events_and_metrics_include_build()
         assertThat(events.get(2).getExecutorCount().orElse(null), is(1));
     }
 
+    @Test
+    @Issue("JENKINS-69817")
+    public void given__a_job_with_subtasks__when__built__then__build_include_subtasks_metrics() throws Exception {
+        MyListener listener = ExtensionList.lookup(QueueItemMetricsListener.class).get(MyListener.class);
+        assertThat(listener, notNullValue());
+        
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "testSubTasks");
+        p.setDefinition(new CpsFlowDefinition("node { sleep 1 }" +
+            "\nnode { sleep 2 }", true));
+//        j.jenkins.setQuietPeriod(0);
+        WorkflowRun workflowRun = Optional.ofNullable(p.scheduleBuild2(0))
+            .orElseThrow(AssertionFailedError::new)
+            .waitForStart();
+        j.waitForCompletion(workflowRun);
+        j.waitUntilNoActivity();
+
+        List<TimeInQueueAction> timeInQueueActions = workflowRun.getActions(TimeInQueueAction.class);
+        assertThat(timeInQueueActions, notNullValue());
+        assertThat(timeInQueueActions, hasSize(1));
+        
+        List<SubTaskTimeInQueueAction> subTaskActions = workflowRun.getActions(SubTaskTimeInQueueAction.class);
+        assertThat(subTaskActions, notNullValue());
```

Comment:
Is this what fails with the `src/main/` change?

Reply:
The current problem is that `SubTaskTimeInQueueAction` sare never added to the builds. The main test that validate the fix is:

```
assertThat(subTaskActions, hasSize(2));
```